loopdb/store.go

```diff
@@ -89,33 +89,39 @@ func NewBoltSwapStore(dbPath string, chainParams *chaincfg.Params) (
 		}
 	}
 
+	path := filepath.Join(dbPath, dbFileName)
+	dbExists := fileExists(path)
+
 	// Now that we know that path exists, we'll open up bolt, which
 	// implements our default swap store.
-	path := filepath.Join(dbPath, dbFileName)
 	bdb, err := bbolt.Open(path, 0600, nil)
 	if err != nil {
 		return nil, err
 	}
 
 	// We'll create all the buckets we need if this is the first time we're
 	// starting up. If they already exist, then these calls will be noops.
-	err = bdb.Update(func(tx *bbolt.Tx) error {
-		_, err := tx.CreateBucketIfNotExists(loopOutBucketKey)
-		if err != nil {
-			return err
-		}
-		_, err = tx.CreateBucketIfNotExists(loopInBucketKey)
-		if err != nil {
-			return err
-		}
-		_, err = tx.CreateBucketIfNotExists(metaBucket)
+	if !dbExists {
+		log.Infof("Initializing new db")
+		err = bdb.Update(func(tx *bbolt.Tx) error {
+			_, err := tx.CreateBucketIfNotExists(loopOutBucketKey)
+			if err != nil {
+				return err
+			}
+			_, err = tx.CreateBucketIfNotExists(loopInBucketKey)
+			if err != nil {
+				return err
+			}
+			_, err = tx.CreateBucketIfNotExists(metaBucket)
+			if err != nil {
+				return err
+			}
+
+			return setDBVersion(tx, latestDBVersion)
```

Comment:
Could be, but then `syncVersions` would iterate through all migrations for an empty database. It is a different way of doing it and it would also mean that we should only create buckets in migrations. Also not quite what we do in `lnd`, because there a lot of `CreateIfNotExists` is going on.